plugins/services/watchdog/service.go

```diff
@@ -58,17 +60,23 @@ func init() {
 type service struct {
 }
 
-func notifySystemd(interval time.Duration) {
-	go func() {
-		ticker := time.NewTicker(interval)
-		defer ticker.Stop()
+func notifyWatchdog(interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
 
-		for range ticker.C {
-			ack, err := daemon.SdNotify(false, daemon.SdNotifyWatchdog)
-			if err != nil {
-				fmt.Println("WATCHDOG ERRROR - ", err)
-			}
-			fmt.Println("Sent watchdog notification -", ack)
+	for range ticker.C {
+		ack, err := daemon.SdNotify(false, daemon.SdNotifyWatchdog)
+		if err != nil {
+			fmt.Println("Failed to notify systemd watchdog - ", err)
 		}
-	}()
+		fmt.Println("Sent watchdog notification -", ack)
+	}
 }
+
+func NotifyReady(){
```

Comment:
@samuelkarp, I'm not sure what level of health checking we want to introduce with this. I added a method to notify system manager with READY status. I did not create the interface yet as it is WIP